src/rules/utils.js

```diff
@@ -19,61 +19,36 @@
 
 import { IntlMessageFormat } from 'intl-messageformat';
 
-function processNode(node) {
-    let text = "";
-    switch (node.type) {
-        case 0:
-            text = node.value;
-            break;
-
-        // Actual parameter
-        case 1:
-            text = `{${node.value}}`;
-            break;
-
-        case 6:
-            if (node.options) {
-                text = Object.keys(node.options).map(category => {
-                    return concatText(node.options[category].value);
-                }).join(" ");
+/** @returns {string} */
+function concatIntlAstText(/** @type {import("@formatjs/icu-messageformat-parser").MessageFormatElement[]} */ astElements) {
+    return astElements
+        .map((element) => {
+            switch (element.type) {
+                case 0: // literal
+                    return element.value;
+                case 1: // argument
+                    return "{" + element.value + "}";
+                case 6: // plural
```

Comment:
For select and selectordinal and for rich-text tags, add in `case 5:` which falls through to case 6, and `case 8:` which recursively adds text from the `children` property.

Reply:
Added in f794a91